engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -489,10 +491,13 @@ private void updateCamera(CharacterMovementComponent charMovementComp, Vector3f
             }
         }
 
-        if (charMovementComp.mode == MovementMode.GHOSTING) {
-            playerCamera.extendFov(24);
-        } else {
-            playerCamera.resetFov();
+        if(charMovementComp.mode != lastMode){
+            if (charMovementComp.mode == MovementMode.GHOSTING) {
```

Comment:
`noclip` is the traditional meaning for a 'ghost' (travel through things) mode in engines, if you are looking for a more specific name.  Where `flying` is the traditional term for being able to fly unbounded.  And both combined is usually not given its own name though some things of recent use `spectator` mode for that.